imageproxy.go

```diff
@@ -185,6 +190,13 @@ func (p *Proxy) serveImage(w http.ResponseWriter, r *http.Request) {
 	if p.FollowRedirects {
 		// FollowRedirects is true (default), ensure that the redirected host is allowed
 		p.Client.CheckRedirect = func(newreq *http.Request, via []*http.Request) error {
+			if p.MaxRedirects > 0 && uint8(len(via)) > p.MaxRedirects {
+				if p.Verbose {
+					p.logf("followed too many redirects: %d", len(via))
+				}
+				http.Error(w, errTooManyRedirects.Error(), http.StatusBadRequest)
```

Comment:
I really don't know what good response status is for this scenario so I wend for the catch-all `400` for now 🤷‍♂️